### ISSUE TITLE
Absolutely awesome

### ISSUE THREAD

**Issue by Noitidart** [MozillaOnline/easyscreenshot#2]
I seriously love this addon. This addon seems to be just Firefox only? Are you firefox fan? You dont plan to make for other browsers? Thats how I am with my addons :D I was going to make a WebPageScreenshot addon that was Mozilla only but it looks like you did that so I would love to collaborate on it I have tons of ideas just chatting here on github would be fun. :D

>
by Wei Deng (Developer) on August 11, 2015 · permalink · translate
The project is on github, you can get it at:https://github.com/MozillaOnline/easyscreenshot
I will fix the light theme problem in the next version, and you also can fix it if you like to do it.
>
As you know, from Fx41 all the addons need to be signed, we would try to avoid some NATIVE code(like jsctypes and npapi plugins), so, do you have any idea how to get cursor moving direction without native code, when exceeds Fx window bound?

Thanks for this reply. npapi is deprecated and we can't do that. I saw that you use a dll file for and then use that via jsctypes, does that require you to update every new fierfox release?

**Comment by weideng** [comment]
Yes, it's for Fx only now, and we still don't have any plan to support other browsers, but you can do it if you like :D 
There is a dll file, for Windows, it's used for capturing desktop screen, it's based on WinAPi, so it's not necessary to update it for every Fx release.

**Comment by Noitidart** [comment]
Noooooooooo firefox onlyyy!!! hahahahah the greatest stuff i want to keep firefox only!! :package: 

No plans ever to help the other browsers!! Well from my side Im looking to collaborate with others like me :p

Ah so the native-xpcom components are what need update every release?

**Comment by weideng** [comment]
native-xpcom components will be deprecated, so you'd better avoid it.

**Comment by Noitidart** [comment]
sweetness i created nativeshot https://addons.mozilla.org/en-US/firefox/addon/nativeshot/

and am waiting for review its taking some time. i was a big fan of LightShot addon which took native screenshot on windows only so i made nativeshot which is pure js-ctypes and does native shot on all os. android 4.* is a problem for me right now, but androind 5.* i got it.

but my canvas editor sucks and i suuuper liked yours.

**Comment by weideng** [comment]
It works in fennec on android5.*? That's really awesome.

**Comment by Noitidart** [comment]
haha thanks i used JNI via js-ctypes :D sooo cool im documenting it here: 
* https://developer.mozilla.org/en-US/docs/Mozilla/JavaScript_code_modules/JNI.jsm
* https://developer.mozilla.org/en-US/docs/Mozilla/JavaScript_code_modules/JNI.jsm/Port_Java_code_to_JNI.jsm

Thanks to the awesome work from @cscott https://github.com/cscott/intent-addon/issues/2

I havent impelemented the android support into the addon yet. On v1.1 im working on fixing "window selector" on linux (almost done, with it, its good on windows and osx though). That why for your addon based on my window selector idea i thought you should add  an "Element Selector" which would be super useful for selecting elements that exceed the scroll bounds.

Im also adding twitter support in v1.1 and other langugages. Btw i noticed you have chineese suport, would you be able to help me get chinese translation for my addon? http://beta.babelzilla.org/projects/p/nativeshot/ tha twould be really cool!

**Comment by Noitidart** [comment]
also your addon is missing print, upload (imgur, twitter, etc) feature :( the print feature is pretty easy this is how i did it: https://github.com/Noitidart/NativeShot/blob/master/bootstrap.js#L812-L841

i can help you with imgur, its real easy. im working on twitter share right now :) http://stackoverflow.com/questions/31951730/twitter-progamatically-add-photo

**Comment by weideng** [comment]
Oh, it's a pity, the product is mainly focused on Chinese users(not only). Maybe you know, we cann't visit youtube, FB, twitter and many google services easily,(blocked by GFW Chinese government's Great Fire Wall ), and we also have a product team, before adding new features,  we will discuss a lot. So, for me, it would take a longer time to do some new feature, but you can do anything you like.
Actually your ideas are really very good, but I will discuss them with Product guys before doing anything.
 
Appreciate you very much!

**Comment by Noitidart** [comment]
Cool thanks so much but please keep this awesomeness Firefox only pretty please!!

HAHAHA that is so funny like Great Wall of China, they have "Great Fire Wall of China" haha!

I support your country in blocking Google, Google steals peoples lives and they profit off of it, and the poor users are lied to and don't even known, aaand Google doesn't share any of those profits with the users. At least Microsoft gives people 10 dollars a month via bing rewards.

I would seriously love to get involved and make this addon real the number 1 screenshoting addon. That element selector idea is unique, no other screnshot addon has it, and i only shared with you guys as i think you are firefox only :)

I badly wan t to fix the scrolling issue on region select. I also am not a fan of opening a new tab for the editor, I like to make the subject the editor as I did in my NativeShot addon.

by the way would you be able to help me please translate NativeShot to Chinees on babelzilla? :)

**Comment by Noitidart** [comment]
Please try the lightshot addon: https://addons.mozilla.org/en-US/firefox/addon/lightshot/

When you click screenshot notice how it freezes the whole screen. I think we should do the same but for the webpage, so any animation gets frozen. People like to catch things. We also need to add a delayed shot feature like NativeShot.

And the editor tools of Lightshot are very awesome, BUT I like ours in Easy Screenshot more. Because ours allow users to use Ctrl + +/- to zoom in or zoom out to their preference, Lightshot editor tools are very small and not friendly users who have poorer vision.

So our editor tools are very nice, but it should not be in the way of the canvas, it should be hovring over it, we need to think of some creative way, like LightShot did it, but something that maintains our user friendliness of the zoomability of the tools.

**Comment by Noitidart** [comment]
Whoops accdintally closed forgive that please, reopened.

**Comment by weideng** [comment]
Sure, I will keep it Fx only.
OK, I will have someone to translate NativeShot for you and will try lightShot later.

When select "Capture Webpage Region", there is a overlay above origin webpage, then, you can select webpage region(also can adjust the rectangle size and position), if you want to keep editor in the same page, you need another overlay, above region selecting overlay, place the editor in that overlay(now, in easyscreenshot, the editor is opened in a new tab)

**Comment by Noitidart** [comment]
Yes sir we'll have to overlay it with a shot, if you guys like the idea we should do it. It's just Lightshot does it that way and I have seen 50k+ users on windows alone, then another 20k from osx very very like that feature. Because it freezes animations on click.

And I thought a little bit about the "delayed shot" feature, we may not need that because we have hotkey support. It was mainly needed in NativeShot because I didn't create a system wide operating system level hotkey. Which I probably should do sometime.

But maybe we can still consider the delay shot, becuse maybe user wants to mouseoever something to get some effect,  and pressing a key will make that affect go away. It's a simple feature we should maybe add it.

I think we need to do some more targeting adversting from the AMO page and addon description, we should advertise this addon as "the best WebPageShot addon" we need to find another catchy word for "WebPageShot" like "WebShot" or something you know what I mean? You already have a user base of 500k users and also you guys worked so well and got canvas very performant. The performance you acheived on canvas is the number one reason why I want to collaborate with you rather then restart from scratch on one with my ideas.


But yep trust me that freeze content on click is a very desired feature, it gives users absolute control. Where do you guys discuss things, I would love to join your discussions and help you land the print-ability, twitter upload, imgur upload, etc. I'm working on more uploads for NativeShot and want to bring it to this addon as well. Sooo many ideas hahaha here were my ideas for a "Webpage Screenshot Addon": https://github.com/Noitidart/ff-addon-ideas/wiki#webshot

We should then change the label on the CustomizableUI.jsm label from "Easy Screenshot" to a short phrase that clearly conveys the message they are going to be taking a screnshot of webpage content. Like for my NativeShot addon, even though my addon name is "NativeShot", on my CustomizableUI buttons I labeled them "Screenshot" (instead of "NativeShot") as "Screenshot" is globally synonomous with caputring whatever is on all your monitors.

We should also discuss maybe modifying the icon of the addon a bit, like in the middle of the "Crop tool" icon for our logo, we should put inside a web page or something to make it clear. And if you dont mind Ill do the same icon for NativeShot but use a a monitor inside the icon.

I want to make NativeShot and this addon sibiling addons. :) I think thats why we should also drop the "Winapi" native screenshot function from "Easy Screenshot". Because it only takes of the single monitor, it's not very good at doing it. We should really advertise this as "the ultimate webpage screenshoting tool".

Thanks very much for the NativeShot translation help.

And suuuuper duper thanks for Firefox only I looove that!! :dancers: 

Forgivme I have sooooo many ideas and so much passion on this subject, I want to make Firefox have the best tools. I won't do any changes without discussing with your team, I respect your team very much for their work for Firefox only and for the prefosionall level of work they did. So I say this stuff above but only with your teams discussion and agreement should we doanything, And I should and will not force anything I will only be apart of the discussion. :)

**Comment by Noitidart** [comment]
Or you don't modify your icon and i'll take your icon and put a monitor inside, whatever you guys decide. I just want to discuss it with you guys. Maybe even rename your addon to be "WebScreenShot" just brainstorming, but we want to make it targetted its the secret of the media/branding industry. We want to make people synonmous with our product. Other users from other browsers should be like "yeah the products for our browsers are not up to part with NativeShot/Easy Screenshot" etc haha

**Comment by Noitidart** [comment]
Oh we also need to work on getting your screenshot tool to work on all webpages, including XUL pages, like about:addons and newtab etc etc. It's very easily doable, especially with the lightshot method which i used in nativeshot too.

**Comment by Noitidart** [comment]
The other plan I had for my WebPageScreenShot addon was to allow focus into a certain iframe/frame, we may want to discuss that to see if we can do it in some creative way. Especially with the Lightshot method of freezing the subject material on click, we'll need to pick, so if user scrolls the edge it will scroll the right viewport.

**Comment by weideng** [comment]
I really like some of your crazy ideas,  unfortunately, I'm too busy these days. We still have some other addons , they are VERY VERY IMPORTANT for Chinese users, like COBA(Chinese Online Banks Assistant).
I encourage you to do something, you can clone and make a new branch of the project, you can rename it to "WebScreenShot", even use the same icon, you can take your branch as your own project, add any feature you like, you can release it on AMO ......

As you know, Fx is upgrading so fast,  we have more than 20 addons to be signed,  tons of maintaining work, e10n will be released in Fx42, still a lot of work to do ...

Now, we don't have any plan to add new features for these addons, but maintaining only, because of short of hands.

If you want to create a new branch, I will pay an eye on it.

**Comment by Noitidart** [comment]
Hhahaha thanks for that I have sooo many ideas so much passion I just want to be apart of a team as I can do it but the performance is from experience which your team has.

Thanks very much actually the "WebScreenShot" is a pretty crappy name haha. Ideally I would like to join your project and do it with your team, as I am a dev that thinks from the user prespective. If I do a branch then users will think they "moving" to another addon which is psychologically a negative feeling for customers in the media/commercial industry. Haha i did a lot of research on how to make users happy with a great product. The happiness of the users will be very beneficial to Firefox as thats where word of mouth starts. They will be like Firefox has Easy Screenshoot addon blah blah haha. Yeah I did tons of mecia/commercial products research hahah

So would it be possible I join your team, so if you don't have time right now I can do the maintaining until you have time in the future. Having too many products also makes it confusing for users on which to use. Your product has the "reputation" already. :)

I'm working on some other collaborations as well so would just join your team/collaboration on this one. And have an addon called Profilist which won the Australis contest last year and I need to finish that but its very interconnected (not fragile)/complex but can't work on that unless I have time to dedicate it, as I have to think through the process and carefully make changes over weeks.

**Comment by Noitidart** [comment]
Oh I don't care if my name shows, all i care is about if Firefox has the top quality products. So please don't worry I am very receptive to team collaboration :)

**Comment by Noitidart** [comment]
But no forcing at all i understand if team is really tight nit not accepting new members. Its just i'm not great with canvas editor performance, and needed help with that. It will probably be easier for this then it was for nativeshot, as I had to deal with DPI of multiple monitors (scaling and stuff) but initially my perf was pretty bad, i dont have much experience with it.

**Comment by Noitidart** [comment]
Hey man thanks so much for your help @wenke signed up for Chinese translation today for NativeShot :)

**Comment by Noitidart** [comment]
Holy moly your addon went to 2million users! Can you please like hide this topic so google chrome or other browser addon makers dont see our awesome ideas and steal them for other browsers :(

**Comment by Noitidart** [comment]
Hey there @weideng I was wondering if your team would be comfortable with advertising NativeShot in the on your listing of Easyscreenshot on AMO?

NativeShot has come along real nicely. I am working on a canvas editor now. I'm hoping I can get this addon featured, would be my first one! It would be awesome if you guys could put a little note saying "If you are looking for an addon to take screenshots of content outside of Firefox tabs, then NativeShot is an addon you might want to check out - https://addons.mozilla.org/en-US/firefox/addon/nativeshot/ "

Thanks so much! I added global hotkey support for windows/mac/linux. So even if Firefox is not focused the screenshot can get triggered. :)
